src/api/providers/sapaicore.ts

```diff
@@ -428,6 +435,94 @@ export class SapAiCoreHandler implements ApiHandler {
 	}
 
 	async *createMessage(systemPrompt: string, messages: Anthropic.Messages.MessageParam[]): ApiStream {
+		if (this.options.sapAiCoreUseOrchestrationMode ?? true) {
+			yield* this.createMessageWithOrchestration(systemPrompt, messages)
+		} else {
+			yield* this.createMessageWithDeployments(systemPrompt, messages)
+		}
+	}
+
+	// TODO: support credentials changes after initial setup
+	private ensureAiCoreEnvSetup(): void {
+		// Only set up once to avoid redundant operations
+		if (this.aiCoreEnvSetup) {
+			return
+		}
+
+		// Validate required credentials
+		if (
+			!this.options.sapAiCoreClientId ||
+			!this.options.sapAiCoreClientSecret ||
+			!this.options.sapAiCoreTokenUrl ||
+			!this.options.sapAiCoreBaseUrl
+		) {
+			throw new Error("Missing required SAP AI Core credentials for orchestration mode. Please check your configuration.")
```

Comment:
```suggestion
			throw new Error("Missing required SAP AI Core credentials. Please check your configuration.")
```
All are required regardless of orchestration mode. Solely the `AICORE_SERVICE_KEY` env var is implicitly used by `@sap-ai-sdk/orchestration` using `@sap-ai-sdk/ai-api` under the hood

Reply:
I agree, 

I changed the error, and made it a func that will be called also before the authentication step that used in createMessageWithDeployments flow